pkg/kubelet/kubelet_pods.go

```diff
@@ -1092,11 +1094,19 @@ func (kl *Kubelet) podKiller() {
 				if err != nil {
 					klog.Errorf("Failed killing the pod %q: %v", runningPod.Name, err)
 				}
-				lock.Lock()
-				killing.Delete(string(runningPod.ID))
-				lock.Unlock()
 			}(apiPod, runningPod)
 		}
+		// clean every 10 minutes
+		if time.Now().Unix()-timeLastClean > 600 {
```

Comment:
Are pod Ids guaranteed to be unique (i.e. is it impossible that two pods would have the same id, lets say `abc`, in the same 10 minute window)? 

For example, imagine we delete pod `abc` at 10:00 am. Then at 10:01, we create another pod which also has id `abc` (this assumes pod IDs are not guaranteed unique). With this change, I don't think we could delete pod `abc` until 10:10am.

Reply:
Good question - I myself is curious to know :-)